jobs/jobs_api_go_sdk.go

```diff
@@ -199,16 +243,16 @@ func prepareJobSettingsForUpdateGoSdk(d *schema.ResourceData, js *JobSettingsRes
 func prepareJobSettingsForCreateGoSdk(d *schema.ResourceData, jc *JobCreateStruct) error {
 	// We always need to add NumWorkers into ForceSendField for the go-sdk client.
 	// Before the go-sdk migration, the field `num_workers` was required, so we always sent it.
-	for _, task := range jc.Tasks {
-		if task.NewCluster != nil {
-			err := updateJobClusterSpec(task.NewCluster, d)
+	for i := range jc.Tasks {
+		if jc.Tasks[i].NewCluster != nil {
+			err := updateJobClusterSpec(d, fmt.Sprintf("task.%d.new_cluster.0", i), jc.Tasks[i].NewCluster)
 			if err != nil {
 				return err
 			}
 		}
 	}
 	for i := range jc.JobClusters {
-		err := updateJobClusterSpec(&jc.JobClusters[i].NewCluster, d)
+		err := updateJobClusterSpec(d, fmt.Sprintf("job_cluster.%d.new_cluster.0", i), &jc.JobClusters[i].NewCluster)
```

Comment:
These changes were necessary for adding a `schema.ResourceData` prefix argument.